lib/api/model/events.dart

```diff
@@ -832,6 +834,71 @@ class UpdateMessageFlagsMessageDetail {
   Map<String, dynamic> toJson() => _$UpdateMessageFlagsMessageDetailToJson(this);
 }
 
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class TypingStatusEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'typing';
+
+  final TypingStatusOp op;
+  final MessageType messageType;
+  final Typist sender;
+  final List<Typist>? recipients;
```

Comment:
Actually, though, instead of having this `Typist` (or `TypingEventUser`) class at all, I think we can simplify by just dropping the emails and making these an `int` and a `List<int>`. Any time you see a "Zulip API email address" mentioned in the docs, that's basically an obsolescent relic of the days when Zulip users were primarily identified by their email address, rather than their immutable numeric user ID. There might still be a few corners of the API where we need to use them; but our first choice should always be to ignore them and see if we can do everything with just the user IDs.

For an example of a similar simplification, see `SubscriptionRemoveEvent.streamIds`.

Reply:
Good thought! This appears to be a good case to use `readValue` for.